src/test/scala/mesosphere/marathon/api/v2/SpecInstancesResourceTest.scala

```diff
@@ -72,19 +72,56 @@ class SpecInstancesResourceTest extends UnitTest with GroupCreation {
 
   "SpecInstancesResource" should {
     "deleteMany" in new Fixture {
-      val appId = "/my/app"
+      val appId = "/my/app".toRootPath
       val host = "host"
-      val toKill = Seq(TestInstanceBuilder.newBuilder(PathId(appId)).addTaskStaged().getInstance())
+      val clock = new SettableClock()
+      val instance1 = TestInstanceBuilder.newBuilderWithLaunchedTask(appId, now = clock.now(), version = clock.now()).addTaskStaged().getInstance()
+      val instance2 = TestInstanceBuilder.newBuilderWithLaunchedTask(appId, now = clock.now(), version = clock.now()).addTaskStaged().getInstance()
+      val toKill = Seq(instance1, instance2)
 
       config.zkTimeoutDuration returns 5.seconds
       taskKiller.kill(any, any, any)(any) returns Future.successful(toKill)
-      groupManager.runSpec(appId.toRootPath) returns Some(AppDefinition(appId.toRootPath))
+      groupManager.runSpec(appId) returns Some(AppDefinition(appId))
+      healthCheckManager.statuses(appId) returns Future.successful(collection.immutable.Map.empty)
 
-      val response = appsTaskResource.deleteMany(appId, host, scale = false, force = false, wipe = false, auth.request)
+      val response = appsTaskResource.deleteMany(appId.toString, host, scale = false, force = false, wipe = false, auth.request)
       response.getStatus shouldEqual 200
+
+      val expected =
+        s"""
+           |{ "tasks": [
+           |  {
+           |    "appId" : "/my/app",
+           |    "healthCheckResults" : [ ],
```

Comment:
According to the raml, `healthCheckResults`, `ipAddresses`, `ports`, `servicePorts` and `localVolumes` are optional fields. It would be consecutive to add `pragma.omitEmpty` so that the fields are left out if empty instead of being rendered as empty arrays.

I'd favor the pragma to be added so the output is less cluttered with empty arrays.

Reply:
Isn't that breaking the API again? Currently the RAML is not our ground truth.

Reply:
Yeah, it is breaking. We can't omit empty fields without a major version bump.

Reply:
If the actual API contract is mandatory, the raml shouldn't state otherwise. If this is the case,
I would be inclined to change the raml. The schema shouldn't say fields are optional, if it is **not** safe to omit them.

However, from what I can see and test, `healthCheckResults`, `ports`, `servicePorts` and `localVolumes` are in fact omitted in 1.5.x if they're not set/empty. I am currently not sure whether it is possible to get a task w/o `ipAddresses` but this shouldn't be hard to test.

So, what is breaking if we use `pragma.omitEmpty` on those that have always been omitted (verified: 1.3.10, 1.4.5, 1.5.2)?